xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp

```diff
@@ -439,8 +383,9 @@ void CPeripheralBusAddon::UpdateAddons(void)
   std::set<std::string> removed;
 
   // Get new add-ons
-  PeripheralAddonVector newAddons = GetEnabledAddons();
-  std::transform(newAddons.begin(), newAddons.end(), std::inserter(newIds, newIds.end()), GetPeripheralAddonID);
+  BinaryAddonBaseList newAddons;
+  CServiceBroker::GetBinaryAddonManager().GetAddonInfos(newAddons, true, ADDON_PERIPHERALDLL);
```

Comment:
This and the other is thought as replacement for the CBinaryAddonCache and to remove them in future, both does mostly the same.